api/client/gro_client_test.py

```diff
@@ -1,92 +1,373 @@
+        'start_date': '2000-02-18T00:00:00.000Z',
+        'frequency_id': 3,
+        'end_date': '2020-03-12T00:00:00.000Z',
+        'name': '8-day'
+    }, {
+        'start_date': '2019-09-02T00:00:00.000Z',
+        'frequency_id': 1,
+        'end_date': '2020-03-09T00:00:00.000Z',
+        'name': 'daily'
+    }]
+
+
+def mock_get_top(access_token, api_host, entity_type, num_results, **selection):
+    return [
+        {'metricId': 860032, 'itemId': 274, 'regionId': 1215, 'frequencyId': 9,
+         'sourceId': 2, 'value': 400, 'unitId': 14},
+        {'metricId': 860032, 'itemId': 274, 'regionId': 1215, 'frequencyId': 9,
+         'sourceId': 2, 'value': 395, 'unitId': 14},
+        {'metricId': 860032, 'itemId': 274, 'regionId': 1215, 'frequencyId': 9,
+         'sourceId': 2, 'value': 12, 'unitId': 14},
+    ]
+
+
+def mock_get_data_points(access_token, api_host, **selections):
+    if isinstance(selections['region_id'], int):
+        print('mock_data_points[0]', mock_data_points[0])
+        return [dict(mock_data_points[0])]
+    elif isinstance(selections['region_id'], list):
+        return [dict(mock_data_points[0]), dict(mock_data_points[1])]
+
+
+@patch('api.client.lib.get_available', MagicMock(side_effect=mock_get_available))
+@patch('api.client.lib.list_available', MagicMock(side_effect=mock_list_available))
+@patch('api.client.lib.lookup', MagicMock(side_effect=mock_lookup))
+@patch('api.client.lib.get_allowed_units', MagicMock(side_effect=mock_get_allowed_units))
+@patch('api.client.lib.get_data_series', MagicMock(side_effect=mock_get_data_series))
+@patch('api.client.lib.search', MagicMock(side_effect=mock_search))
+@patch('api.client.lib.rank_series_by_source', MagicMock(side_effect=mock_rank_series_by_source))
+@patch('api.client.lib.get_geo_centre', MagicMock(side_effect=mock_get_geo_centre))
+@patch('api.client.lib.get_geojson', MagicMock(side_effect=mock_get_geojson))
+@patch('api.client.lib.get_descendant_regions', MagicMock(side_effect=mock_get_descendant_regions))
+@patch('api.client.lib.get_available_timefrequency', MagicMock(side_effect=mock_get_available_timefrequency))
+@patch('api.client.lib.get_top', MagicMock(side_effect=mock_get_top))
+@patch('api.client.lib.get_data_points', MagicMock(side_effect=mock_get_data_points))
+class GroClientTests(TestCase):
+    def setUp(self):
+        self.client = GroClient(MOCK_HOST, MOCK_TOKEN)
+        self.assertTrue(isinstance(self.client, GroClient))
+
+    def test_get_logger(self):
+        # should NOT raise any exception if get_logger correctly returns a logger object:
+        self.client.get_logger().debug('Test output')
+
+    def test_get_available(self):
+        self.assertTrue('name' in self.client.get_available('units')[0])
+
+    def test_list_available(self):
+        self.assertTrue('metric_id' in self.client.list_available({})[0])
+
+    def test_lookup(self):
+        self.assertEqual(self.client.lookup('units', 10)['name'], 'kilogram')
+
```

Comment:
minro, missing tests for search_and_lookup, lookup_belongs

Reply:
? They're there:

* https://github.com/gro-intelligence/api-client/pull/218/files#diff-dca7837b9521ac58ccaf2591db344cb5R223
* https://github.com/gro-intelligence/api-client/pull/218/files#diff-dca7837b9521ac58ccaf2591db344cb5R229

Reply:
sorry, mustve misssed them!